SDL2 with Vulkan does not actually have enabled Vulkan on macOS.

**Describe the bug**
- SDL2 with Vulkan does not actually have Vulkan on macOS.
- The SDL video driver does not have Vulkan_CreateSurface and other functions loaded.
- Apparently, because SDL_VIDEO_VULKAN is not defined and this block of code is never invoked.
```
// ... sdl2/src/video/cocoa/SDL_cocoavideo.m
#if SDL_VIDEO_VULKAN
    device->Vulkan_LoadLibrary = Cocoa_Vulkan_LoadLibrary;
    device->Vulkan_UnloadLibrary = Cocoa_Vulkan_UnloadLibrary;
    device->Vulkan_GetInstanceExtensions = Cocoa_Vulkan_GetInstanceExtensions;
    device->Vulkan_CreateSurface = Cocoa_Vulkan_CreateSurface;
    device->Vulkan_GetDrawableSize = Cocoa_Vulkan_GetDrawableSize;
#endif
```

**Environment**
- OS: macOS 10.15.6
- Compiler: Apple clang version 11.0.3 (clang-1103.0.32.62), Target: x86_64-apple-darwin19.6.0
- Vulkan SDK: vulkansdk-macos-1.2.148.1

**To Reproduce**
Steps to reproduce the behavior:
1. vcpkg install sdl2[vulkan]:x64-osx
2. Try creating a window with the SDL_WINDOW_VULKAN flag set.

**Expected behavior**
A created window, instead of nullptr.

**Failure logs** 
- N/A

**Additional context**
- N/A

This bug recently bit me as well. I did a bit of digging. The cmake for the sdl project is failing the test for metal. From config-x64-osx-dbg-out.log in the buildtree:
```
-- Looking for dlopen
-- Looking for dlopen - found
-- Performing Test HAVE_FRAMEWORK_METAL
-- Performing Test HAVE_FRAMEWORK_METAL - Failed
-- Performing Test HAVE_PTHREADS
-- Performing Test HAVE_PTHREADS - Success
```
This disables vulkan support as well. Looking at x64-osx-dbg/CMakeFiles/CMakeError.log (again from the buildtree):
```
Performing C SOURCE FILE Test HAVE_FRAMEWORK_METAL failed with the following output:
Change Dir: /Users/xxx/buildtrees/sdl2/x64-osx-dbg/CMakeFiles/CMakeTmp

Run Build Command(s):/Users/xxx/downloads/tools/ninja-1.10.1-osx/ninja cmTC_41e88 && [1/2] Building C object CMakeFiles/cmTC_41e88.dir/src.c.o
[2/2] Linking C executable cmTC_41e88
FAILED: cmTC_41e88 
: && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -fPIC -idirafter "/Users/xxx/buildtrees/sdl2/src/SDL2-2-67b29cc446.clean/src/video/khronos" -DHAVE_FRAMEWORK_METAL  -x objective-c -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/cmTC_41e88.dir/src.c.o -o cmTC_41e88   && :
CMakeFiles/cmTC_41e88.dir/src.c.o:1:1: error: source file is not valid UTF-8
<CF><FA><ED><FE><U+0007><U+0000><U+0000><U+0001><U+0003><U+0000><U+0000><U+0000><U+0001><U+0000><U+0000><U+0000><U+0004><U+0000><U+0000><U+0000><U+0008><U+0002><U+0000><U+0000><U+0000> <U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0019><U+0000><U+0000><U+0000><88><U+0001><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>x<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>(<U+0002><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>x<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0007><U+0000><U+0000><U+0000><U+0007><U+0000><U+0000><U+0000><U+0004><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__text<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__TEXT<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+000F><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>(<U+0002><U+0000><U+0000><U+0004><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0004><U+0000><80><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__objc_imageinfo__DATA<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+000F><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0008><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>7<U+0002><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0010><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__compact_unwind__LD<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0018><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000> <U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>@<U+0002><U+0000><U+0000><U+0003><U+0000><U+0000><U+0000><A0><U+0002><U+0000><U+0000><U+0001><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0002><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__eh_frame<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__TEXT<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>8<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>@<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>`<U+0002><U+0000><U+0000><U+0003><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+000B><U+0000><U+0000>h<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>2<U+0000><U+0000><U+0000><U+0018><U+0000><U+0000><U+0000><U+0001><U+0000><U+0000><U+0000><U+0000><U+000F>
^
CMakeFiles/cmTC_41e88.dir/src.c.o:1:2: error: source file is not valid UTF-8
<CF><FA><ED><FE><U+0007><U+0000><U+0000><U+0001><U+0003><U+0000><U+0000><U+0000><U+0001><U+0000><U+0000><U+0000><U+0004><U+0000><U+0000><U+0000><U+0008><U+0002><U+0000><U+0000><U+0000> <U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0019><U+0000><U+0000><U+0000><88><U+0001><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>x<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>(<U+0002><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>x<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0007><U+0000><U+0000><U+0000><U+0007><U+0000><U+0000><U+0000><U+0004><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__text<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__TEXT<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+000F><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>(<U+0002><U+0000><U+0000><U+0004><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0004><U+0000><80><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__objc_imageinfo__DATA<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+000F><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0008><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>7<U+0002><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0010><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__compact_unwind__LD<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0018><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000> <U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>@<U+0002><U+0000><U+0000><U+0003><U+0000><U+0000><U+0000><A0><U+0002><U+0000><U+0000><U+0001><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0002><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__eh_frame<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>__TEXT<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>8<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>@<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>`<U+0002><U+0000><U+0000><U+0003><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+000B><U+0000><U+0000>h<U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000><U+0000>2<U+0000><U+0000><U+0000><U+0018><U+0000><U+0000><U+0000><U+0001><U+0000><U+0000><U+0000><U+0000><U+000F>
```
Goes on like that for a few hundred lines and then:
```
fatal error: too many errors emitted, stopping now [-ferror-limit=]
471 warnings and 20 errors generated.
ninja: build stopped: subcommand failed.

Source file was:

        #include <AvailabilityMacros.h>
        #import <Metal/Metal.h>
        #import <QuartzCore/CAMetalLayer.h>

        #if TARGET_OS_SIMULATOR || (!TARGET_CPU_X86_64 && !TARGET_CPU_ARM64)
        #error Metal doesn't work on this configuration
        #endif
        int main()
        {
            return 0;
        }
```
Which kind of looks like it is trying to compile an object file as a source file. I've attached the log files.

[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/5498065/config-x64-osx-dbg-out.log)
[CMakeError.log](https://github.com/microsoft/vcpkg/files/5498067/CMakeError.log)

--- a/ports/sdl2/0008-fix-macos-metal-test.patch
+++ b/ports/sdl2/0008-fix-macos-metal-test.patch
@@ -1,0 +1,11 @@
+--- CMakeLists.txt.orig	2020-11-05 16:48:40.000000000 -0800
++++ CMakeLists.txt	2020-11-05 16:49:02.000000000 -0800
+@@ -1701,7 +1701,7 @@
+ 
+     if(VIDEO_VULKAN OR VIDEO_METAL OR RENDER_METAL)
+       set(ORIG_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+-      set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -x objective-c")
++      set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -ObjC")
+       check_c_source_compiles("
+         #include <AvailabilityMacros.h>
+         #import <Metal/Metal.h>

--- a/ports/sdl2/portfile.cmake
+++ b/ports/sdl2/portfile.cmake
@@ -17,6 +17,7 @@ vcpkg_extract_source_archive_ex(
         fix-EventToken-header-reference.patch
         0006-sdl2-Enable-creation-of-pkg-cfg-file-on-windows.patch
         0007-sdl2-skip-ibus-on-linux.patch
+        0008-fix-macos-metal-test.patch
 )
 
 string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" SDL_STATIC)

--- a/ports/sdl2/vcpkg.json
+++ b/ports/sdl2/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "sdl2",
   "version-string": "2.0.12",
-  "port-version": 6,
+  "port-version": 7,
   "description": "Simple DirectMedia Layer is a cross-platform development library designed to provide low level access to audio, keyboard, mouse, joystick, and graphics hardware via OpenGL and Direct3D.",
   "homepage": "https://www.libsdl.org/download-2.0.php",
   "features": {

